bundles/AdminBundle/Controller/Admin/Asset/AssetController.php

```diff
@@ -2477,6 +2477,14 @@ public function importServerFilesAction(Request $request)
 
         $files = explode('::', $request->get('files'));
 
+        // prevent directory traversal
+        foreach ($files as $file) {
+            $absolutePath = realpath($serverPath.$file);
+            if (!str_starts_with($absolutePath, rtrim(str_replace('../', '', $serverPath), './'))) {
```

Comment:
```suggestion
            if (!str_starts_with($absolutePath, $serverPath)) {
```
I think we don't need the string manipulation for `$serverPath`, because the realpath is stored in the variable https://github.com/pimcore/pimcore/pull/15664/files#diff-8a94426359af1cadbcb7fdeb7af5731ce2ac295a758ed07a38d36db9d5d85ff9R2473 
What do you think?

Reply:
@robertSt7 thx for the review and your feedback, you are right, i removed the string manipulation check